add a top-level !user command with subcommands

ryanlerch opened the pull request:
This replaces the old !user command (which displayed user info) with a new top level !user command with the following sub-commands (localtime, hello, info).

Had to implement aliases for these 3 subcommands manually, as maubot wont let you alias a command-level to a command + subcommand. So these are done with a passive command.

abompard commented on fedora/fas.py:
```diff
@@ -100,19 +100,7 @@ async def group_info(self, evt: MessageEvent, groupname: str) -> None:
-           (e.g. @username:fedora.im)
-
-        """
+    async def _user_hello(self, evt: MessageEvent, username: str) -> None:
```
If `username` can be `None`, let's state it in the type hints: `username: str | None`

abompard commented on fedora/fas.py:
```diff
@@ -128,19 +116,7 @@ async def hello(self, evt: MessageEvent, username: str) -> None:
-           (e.g. @username:fedora.im)
-
-        """
+    async def _user_info(self, evt: MessageEvent, username: str) -> None:
```
If `username` can be `None`, let's state it in the type hints: `username: str | None`

ryanlerch commented on fedora/fas.py:
```diff
@@ -158,22 +134,10 @@ async def user(self, evt: MessageEvent, username: str) -> None:
         await evt.mark_read()
         try:
-            user = await get_fasuser(username, evt, self.plugin.fasjsonclient)
+            user = await get_fasuser(username or evt.sender, evt, self.plugin.fasjsonclient)
```
heh -- i kinda like that option :)

moreso the other user commands default to using the sender as the default, so i figured it wouldnt hurt to do it here too.

ryanlerch commented on tests/test_bot.py:
```diff
@@ -17,9 +17,7 @@ async def test_help(bot, plugin):
-        "● `!hello [username]` - Return brief information about a Fedora user.\n"
-        "● `!localtime <username>` - Returns the current time of the user.\n"
-        "● `!user [username]` - Return brief information about a Fedora user.\n"
+        "● `!user  <subcommand> [...]` - Get information about Fedora Accounts users\n"
```
i was a little worried that the aliases were starting to pollute the top - level commands list. we still have !user hello as the main command.

Adding to my todolist to try to figure out a neat way to get alias help here.

ryanlerch commented:
Okay, fixed:

* the type checking issues on the username params
* added the r"" string prefix on the regex

other comments i have commented back with my thoughts
